server/routes/notification.js

```diff
@@ -0,0 +1,224 @@
+    const userLists = await UserListModel.findAll({
+      where: {
+        id: listProds.map((listProd) => {
+          return listProd.list_id;
+        }),
+      },
+      transaction: trans,
+    });
+
+    //For each of the over all users in the userLists, we build the data object which will be
+    //placed inside the NotificationModel and its other attributes. This iterates for each of the userLists, but conditionals
+    //are placed to only make one per user. If the model a schema has already been made for a user, we instead
+    //make an update to the objects data.listLocations
+    userLists.forEach((userList) => {
+      //If we havent made a data object for the user, make one.
+      if (!newNotifs[`${userList.user_id}`]) {
+        const data = {
+          title,
+          productId,
+          price,
+          previousPrice,
+          listLocations: [userList.id],
+        };
+
+        newNotifs[`${userList.user_id}`] = {
+          type: "price",
+          data,
+          user_id: userList.user_id,
+          isRead: false,
+        };
+
+        //The data object has an array listLocations. This is basically the lists of a user where the
+        //particular product is placed. If a data object is already made, we simply add in the current list
+        //to the listLocations array.
+      } else {
+        newNotifs[`${userList.user_id}`].data.listLocations.push(userList.id);
+      }
+    });
+
+    //This would convert the newNotifs object into an array.
+    //[ [dataObject], [dataObject], ... ]
+    //Inside the data object, the [0] index is the key, while the [0] index is the data object itself.
+    const parsedNotifs = Object.entries(newNotifs);
+
+    const notifPromiseArr = [];
+
+    //This iteration is the main part of creating the notification
+    parsedNotifs.forEach((notif) => {
+      notifPromiseArr.push(
+        NotificationModel.create(notif[1], { transaction: trans })
+      );
+    });
+
+    //This is the array of all the new notifications made for a user.
+    //May be changed in the future to just [1] for success since no one need to know
+    const allNewNotifs = await Promise.all(notifPromiseArr);
+
+    //Update Product Model
+    await ProductModel.update(
+      {
+        current_price: price,
+        previous_price: productModel.current_price,
+      },
+      {
+        where: {
+          id: productId,
+        },
+        transaction: trans,
+      }
+    );
+
+    await trans.commit();
+
+    res.status(201).send(allNewNotifs);
+  } catch (error) {
+    await trans.rollback()
+    console.error(error);
+    res.send({
+      error: {
+        msg: "Server Error while using service",
+        data: error,
+      },
+    });
+  }
+});
+
+//Find specifically ALL notifications of a user about a price change
+//Accepts "order" as a query. Be default is "DESC" to show latest notifs first
+router.get("/price/all", authMiddleware, async (req, res) => {
```

Comment:
we're soon going to be making notifications for `follows` as well. Maybe this route and the next would make more sense to not be price specific and return all notification types. Maybe in the future we can add the ability to use a query to only get a certain type of notification

Reply:
I could maybe just keep the code and make another route for fetching all notifications of the user.  This is just incase in the future wherein if we have a notifications page, users can filter out which notifications they want see

Reply:
Added another route for getting user notifications with query properties for pagination.